src/v/cluster/scheduling/types.h

```diff
@@ -360,4 +360,31 @@ struct allocation_request {
     friend std::ostream& operator<<(std::ostream&, const allocation_request&);
 };
 
+/**
+ *  The raw_allocation_request represents a simplified allocation_request that
+ *  doesn't allow for partition-specific requirements.
+ *  The memory required for a raw_allocation_request objects does not scale
+ *  with the number of requested partitions.
+ */
+struct raw_allocation_request {
```

Comment:
nit: maybe we can name that `simple_allocation_request` ?
Also consider separating a commit in which a request is added from the commit in which it is used